### PR TITLE
[FLINK-23120][python] Fix ByteArrayWrapperSerializer.serialize to use writeInt to serialize the length

### PR DESCRIPTION
**Pull request by dianfu**
## What is the purpose of the change

*This pull request fixes ByteArrayWrapperSerializer*

## Verifying this change

This change added tests and can be verified as follows:

  - *Added tests ByteArrayWrapperSerializerTest*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-python/src/test/java/org/apache/flink/streaming/api/utils/ByteArrayWrapperSerializerTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    @Override
+    protected int getLength() {
+        return -1;
+    }
+
+    @Override
+    protected ByteArrayWrapper[] getTestData() {
+        return new ByteArrayWrapper[] {
+            randomByteArray(),
```
Comment (StephanEwen) [inline flink-python/src/test/java/org/apache/flink/streaming/api/utils/ByteArrayWrapperSerializerTest.java:50]: I think it is better to have the critical cases as dedicated data. The test should include at least one empty array, one short length, one long length (> 256 bytes) and one very long (> 64k). Random arrays can be added, but the important ones should always be in there.

Comment (dianfu) [reply flink-python/src/test/java/org/apache/flink/streaming/api/utils/ByteArrayWrapperSerializerTest.java]: It makes sense to add dedicated data test. Regarding to 64k length, it fails with OutOfMemoryError 1 times every two runs in my desktop and so I change it to 32k (succeed running the test case 10 times) for test stability.

**Comment by dianfu** [comment]
@StephanEwen Thanks a lot for the review. Have updated the PR for the test cases.

**Comment by dianfu** [comment]
@flinkbot run azure

**Review by HuangXingBo — APPROVED** [review]

**Comment by dianfu** [comment]
@StephanEwen @HuangXingBo Thanks a lot for the review. I'm merging this PR to catch up with the bugfix release of 1.13 as this is a series bug for MapState in Python DataStream API. Feel free to let me know if there is anything we could improve.

**Pull request closed without merging**